calculator_test.go

```diff
@@ -2,23 +2,200 @@ package calculator_test
 
 import (
 	"calculator"
+	"fmt"
+	"math/rand"
 	"testing"
 )
 
+type testCase struct {
+	a, b        float64
+	want        float64
+	name        string
+	errExpected bool
+}
+
+type variadicTestCase struct {
+	nums        []float64
+	want        float64
+	name        string
+	errExpected bool
+}
+
+type evaluateTestCase struct {
+	expression  string
+	want        float64
+	errExpected bool
+}
+
 func TestAdd(t *testing.T) {
 	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
-	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+	testCases := []variadicTestCase{
+		{nums: []float64{1, 2, 3}, want: 6, name: "1+2+3 = 6"},
+		{nums: []float64{1, 2, 3, 4}, want: 10, name: "1+2+3+4 = 10"},
+	}
+
+	for _, tc := range testCases {
+		got := calculator.Add(tc.nums...)
+
+		if tc.want != got {
+			t.Errorf("Error: unexpected value returned - want %.1f, got %.1f", tc.want, got)
```

Comment:
We could say both less and more here. We don't need to say 'Error': that's implied, because if you see any message at all during a test run, it's a failure. Similarly, we don't need to say 'unexpected value'. But we would like to see the _inputs_ which caused the failure, so that we can quickly identify which test case is failing (we could also print the name of the test case).

Reply:
Makes sense, this, along with all the others, is addressed and fixed in dfa85e1